src/development/accessibility-and-localization/internationalization.md

```diff
@@ -264,9 +330,227 @@ return MaterialApp(
       DemoLocalizations.of(context).title,
 ```
 
-For more information about the localization tool,
-such as dealing with DateTime and handling plurals,
-see the [Internationalization User's Guide][].
+### Placeholders, plurals, and selects
+You can also include application values in a message with
+special syntax that uses a _placeholder_ to generate a method
+instead of a getter.
+A placeholder, which must be a valid Dart identifier name,
+becomes a positional parameter in the generated method in the
+`AppLocalizations` code. Define a placeholder name by wrapping
+it in curly braces as follows:
+
+```json
+"{placeholderName}"
+```
+
+Define each placeholder in the `placeholders` object in the app's `.arb` file.
+For example, to define a hello message with a `userName` parameter,
+add the following to `lib/l10n/app_en.arb`:
+
+<?code-excerpt "gen_l10n_example/lib/l10n/app_en.arb" skip="5" take="10" replace="/},$/}/g"?>
+```json
+"hello": "Hello {userName}",
+"@hello": {
+  "description": "A message with a single parameter",
+  "placeholders": {
+    "userName": {
+      "type": "String",
+      "example": "Bob"
+    }
+  }
+}
+```
+
+Regenerate the `AppLocalizations` file. This adds a `hello` method call to
+the `AppLocalizations.of(context)` object, and the method accepts
+a parameter of type `String`; the `hello` method returns a string.
+
+Implement this by replacing the code passed into `Builder`
+with the following:
+
+<?code-excerpt "gen_l10n_example/lib/main.dart (Placeholder)" remove="/wombat|Wombats|he'|they|pronoun/"?>
+```dart
+// Examples of internationalized strings.
+return Column(
+  children: <Widget>[
+    // Returns 'Hello John'
+    Text(AppLocalizations.of(context)!.hello('John')),
+  ],
+);
+```
+
+You can also use numerical placeholders to specify multiple values.
+Different languages have different ways to pluralize words.
+The syntax also supports specifying _how_ a word should be pluralized.
+A _pluralized_ message must include a `num` parameter indicating
+how to pluralize the word in different situations.  English, for example,
+pluralizes "person" to "people", but that doesn't go far enough. The
+`message0` plural might be "no people" or "zero people".
+The `messageFew` plural might be
+"several people", "some people", or "a few people". 
+The `messageMany` plural might
+be  "most people" or "many people", or "a crowd". 
+Only the more general `messageOther` field is required.
+The following example shows what options are available:
+
+```json
+"{countPlaceholder, plural, =0{message0} =1{message1} =2{message2} few{messageFew} many{messageMany} other{messageOther}}"
```

Comment:
@johnpryan @khanhnwin @thkim1011  This line is causing our link checker to choke. Apparently. I'd like to merge this PR ASAP, before we freeze the site. Will adding a "skip" message before this JSON code block be enough to fix it?

Reply:
From the error message, the issue seems to be caused a couple more lines down by this line:
```
<?code-excerpt "gen_l10n_example/lib/l10n/app_en.arb" skip="15" take="10" replace="/{{/{{ '{{' }}/g;/},$/}/g"?>
```
which is strange since this wasn't causing an issue as of Friday.

Reply:
I added this replacement in the code-excerpt because Jekyll reserves the pattern `{{ }}` as its own special syntax, and the only way to use the characters `{{` in the documentation is to replace it with `{{ '{{' }}`.